lib/loader/loader.c

```diff
@@ -267,16 +267,15 @@ struct evmc_vm* evmc_load_and_configure(const char* config, enum evmc_loader_err
     if (!vm)
         return NULL;
 
-    if (vm->set_option == NULL && strlen(options) != 0)
-    {
-        ec = set_error(EVMC_LOADER_INVALID_OPTION_NAME, "%s (%s) does not support any options",
-                       vm->name, path);
-        goto exit;
-    }
-
-
     while (strlen(options) != 0)
     {
+        if (vm->set_option == NULL)
+        {
+            ec = set_error(EVMC_LOADER_INVALID_OPTION_NAME, "%s (%s) does not support any options",
+                           vm->name, path);
+            goto exit;
```

Comment:
Could be `break;` now, but maybe keeping `goto` is fine for consistency.

Reply:
I use `goto exit` because the later `switch` is also using it in case of error.